queries/ada/highlights.scm

```diff
@@ -0,0 +1,176 @@
+;; highlight queries.
+;; See the syntax at https://tree-sitter.github.io/tree-sitter/using-parsers#pattern-matching-with-queries
+;; See also https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md#parser-configurations
+;; for a list of recommended @ tags, though not all of them have matching
+;; highlights in neovim.
+
+[
+   "abort"
+   "abs"
+   "abstract"
+   "accept"
+   "access"
+   "all"
+   "array"
+   "at"
+   "begin"
+   "declare"
+   "delay"
+   "delta"
+   "digits"
+   "do"
+   "end"
+   "entry"
+   "exit"
+   "generic"
+   "interface"
+   "is"
+   "limited"
+   "null"
+   "of"
+   "others"
+   "out"
+   "pragma"
+   "private"
+   "range"
+   "synchronized"
+   "tagged"
+   "task"
+   "terminate"
+   "until"
+   "when"
+] @keyword
+[
+   "aliased"
+   "constant"
+   "renames"
+] @storageclass
+[
+   "mod"
+   "new"
+   "protected"
+   "record"
+   "subtype"
+   "type"
+] @keyword.type
```

Comment:
this is a new capture. but as a subcapture of course allowed

Reply:
Let's keep it for now, and we can change it if/when treesitter proposes a standard approach for this one. In a strongly typed language like Ada, it is kind of nice to highlight type definition statements in a different color (in my daily neovim usage, this is the main difference between the regexp-based highlighting and the treesitter one).

Reply:
We once had `@structure` which was specifically for highlighting the `struct` keyword in C. It has a built-in highlight but I don't think it got added again.

Reply:
Perhaps I should use @keyword.typedef to make it clearer that it is to define a type, and not an actual type (though a keyword would not be a type anyway)